src/libcore/tests/test_xml.py

```diff
@@ -298,3 +298,166 @@ def test20_upgrade_tree(variant_scalar_rgb):
                                <float name="intIOR" value="1.33"/>
                            </bsdf>
                        </scene>""")
+
+@fresolver_append_path
+def test21_xml_save_plugin(variant_scalar_rgb):
+    from mitsuba.core import xml
+    from mitsuba.python.dict_to_xml import WriteXML
+    import os
+    abs_path = os.path.abspath('resources/data/scenes/dict/dict.xml')
+    writer = WriteXML(abs_path)
+    scene_dict = {
+            "type": "sphere",
+            "center" : [0, 0, -10],
+            "radius" : 10.0,
+        }
+    writer.configure(scene_dict)
+    s1 = xml.load_dict({
+        'type': 'scene',
+        'sphere':{
+            "type": "sphere",
+            "center" : [0, 0, -10],
+            "radius" : 10.0,
+        }
+        })
+    s2 = xml.load_file(abs_path)
+    assert str(s1) == str(s2)
```

Comment:
It would be great to test the output string directly, the actual XML code produced by the `dict_to_xml` function. Who knows what kind of magic happens in the `xml.load_file`, this extra indirection might hide some implementation bug on the `dict_to_xml` side.